### PR TITLE
Add ardupilot dockerfile

### PR DESCRIPTION
**Pull request by Thomas-Neill**
Adds ardupilot sitl to the dockerfile.

### CODE REVIEW

**Review by EricPedley — CHANGES_REQUESTED** [review]
Works on my machine. Can u respond to some of the questions I posted?

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,58 @@
...
+
+VOLUME ["/root/ros2_ws/src/uavf_2024"]
+
+RUN apt-get update
+
+# opencv dependencies (https://stackoverflow.com/questions/55313610/importerror-libgl-so-1-cannot-open-shared-object-file-no-such-file-or-directo)
+RUN apt-get install -y ffmpeg libsm6 libxext6
+
+# utils (not strictly necessary)
+RUN apt-get install -y tmux vim
```
[the remark this answers is not in the document]
Comment (Thomas-Neill) [reply Dockerfile:11]: i learned from the best

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,58 @@
...
-RUN pip install -e .
-
-CMD ["/bin/bash"]
-
-
-# docker build -t epedley/uavf_2024-dev .
\ No newline at end of file
+FROM ardupilot/ardupilot-dev-ros
+
+VOLUME ["/root/ros2_ws/src/uavf_2024"]
```
[the remark this answers is not in the document]
Comment (Thomas-Neill) [reply Dockerfile:3]: Yeah, this doesn't need to be here any more

File: Dockerfile [hunk 3]
```diff
@@ -1,23 +1,58 @@
...
+RUN rosdep install --rosdistro ${ROS_DISTRO} --from-paths src -i -r -y
+RUN bash -c "source /opt/ros/humble/setup.bash && colcon build --cmake-args -DBUILD_TESTING=ON"
+
+RUN echo 'source /opt/ros/humble/setup.bash' >> ~/.bashrc
+RUN echo 'source /root/ros2_ws/install/setup.bash' >> ~/.bashrc
+
+RUN echo 'export PATH=/root/ros2_ws/src/ardupilot/Tools/autotest:$PATH' >> ~/.bashrc
+RUN echo 'export PATH=/usr/lib/ccache:$PATH' >> ~/.bashrc
+
+RUN sudo apt-get install -y python3-dev python3-opencv python3-wxgtk4.0 python3-pip python3-matplotlib python3-lxml python3-pygame
```
[the remark this answers is not in the document]
Comment (Thomas-Neill) [reply Dockerfile:50]: It was what the mavproxy install instructions said: https://ardupilot.org/mavproxy/docs/getting_started/download_and_installation.html

**Review by EricPedley — APPROVED** [review]

**Pull request merged**